chain-impl-mockchain/src/ledger/ledger.rs

```diff
@@ -991,6 +991,60 @@ mod tests {
         }
     }
 
+    #[derive(Clone)]
+    pub struct ArbitraryEmptyLedger(Ledger);
+
+    impl Arbitrary for ArbitraryEmptyLedger {
+        fn arbitrary<G: Gen>(g: &mut G) -> Self {
+            let mut ledger = Ledger::empty(
+                Arbitrary::arbitrary(g),
+                Arbitrary::arbitrary(g),
+                Arbitrary::arbitrary(g),
+                Arbitrary::arbitrary(g),
+            );
+
+            ledger.date = Arbitrary::arbitrary(g);
+            ledger.chain_length = Arbitrary::arbitrary(g);
+            ArbitraryEmptyLedger(ledger)
+        }
+    }
+
+    impl fmt::Debug for ArbitraryEmptyLedger {
+        fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+            f.debug_struct("Ledger")
+                .field("chain_length", &self.0.chain_length())
+                .field("settings", &self.0.settings)
+                .field("date", &self.0.date())
+                .field("era", &self.0.era())
+                .field("static_params", &self.0.get_static_parameters().clone())
+                .finish()
+        }
+    }
+
+    impl Into<Ledger> for ArbitraryEmptyLedger {
+        fn into(self) -> Ledger {
+            self.0
+        }
+    }
+
+    #[quickcheck]
+    pub fn apply_empty_block_prop_test(
+        context: HeaderContentEvalContext,
+        ledger: ArbitraryEmptyLedger,
+    ) -> TestResult {
+        let ledger: Ledger = ledger.into();
+        let should_succeed =
+            context.chain_length == ledger.chain_length.next() && context.block_date > ledger.date;
```

Comment:
is this ever true in the property test?

Reply:
i don't fully understand. if we apply empty block then both conditions are the only conditions determine success. On the other hand should_succedd is only try if arbitrary ledger and arbitrary context have this combination on chain_length and block_data, or am I missing something?

Reply:
that looks fine, although I had to read this thrice